doctor_visits/delphi_doctor_visits/update_sensor.py

```diff
@@ -86,12 +87,16 @@ def update_sensor(
     # as of 2020-05-11, input file expected to have 10 columns
     # id cols: ServiceDate, PatCountyFIPS, PatAgeGroup, Pat HRR ID/Pat HRR Name
     # value cols: Denominator, Covid_like, Flu_like, Flu1, Mixed
+    filename = Path(filepath).name
     data = pd.read_csv(
         filepath,
-        usecols=Config.FILT_COLS,
         dtype=Config.DTYPES,
-        parse_dates=[Config.DATE_COL],
     )
+    logger.info(f"Starting processing {filename} ")
+    data.rename(columns=Config.DEVIANT_COLS_MAP, inplace=True)
```

Comment:
There is reason to believe that [the DataFrame column renaming operation might be slow even w/ `inplace=True`](https://stackoverflow.com/questions/64264211/why-is-renaming-columns-in-pandas-so-slow), especially because all of the indicators code is stuck on Pandas 1.5.3 [for silly reasons](https://github.com/cmu-delphi/covidcast-indicators/issues/1972#issuecomment-2197688725).  An alternative to [renaming the columns](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/modify_claims_drops.py#L32-L41) would be to adjust [the column name "constants"](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/config.py#L23-L28) depending on whichever are present in the input file.

You removed one read->rename->write cycle for the source data in `modify_and_write()`, but the file is still read 12 times here ([once for each combination in the loop over 6 geo_types and 2 "weekdays"](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/run.py#L103-L109))...  So moving the renaming to here could be compounding the problem, thus explaining the longer runtime you saw.  Additionally, doing the column filtering and date parsing outside of `read_csv()`, and the column filtering after the `rename()` could also have performance hits.  The only way to know for sure would be to time the individual pandas operations.

Regardless, i think that we can save a lot of running time by moving stuff to outside of `update_sensor()` and outside the geo_type+weekday loop: 
- [file reading and some post-processing (including generating the Weekday params)](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/update_sensor.py#L90-L149) only needs to be done once
- then the [geo_id-specific stuff](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/update_sensor.py#L150-L176) and file outputting will be all thats left inside the loop.

Im not sure why we are [not](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/ansible/templates/doctor_visits-params-prod.json.j2#L14) using the [parallelized version of the geo_id-specific processing](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/update_sensor.py#L179-L208), but i have to assume that (a) it doesnt work or (b) it wasnt fully tested or (c) its slow due to having to copy the big dataframe to all of the spawned threads or (d) it mightve even caused OOMs.

For even more optimizations, [a data copy inside the `Weekday` class](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/_delphi_utils_python/delphi_utils/weekday.py#L144) can be avoided, and something tells me [`write_to_csv()`](https://github.com/cmu-delphi/covidcast-indicators/blob/84d059751b646c0075f1a384741f2c1d80981269/doctor_visits/delphi_doctor_visits/update_sensor.py#L27-L67) could be improved a good bit.